server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/TaintIssueDownloader.java

```diff
@@ -124,13 +124,13 @@ private static ServerTaintIssue convertTaintVulnerability(SourceApi sourceApi, I
       return null;
     }
     var ruleDescriptionContextKey = taintVulnerabilityFromWs.hasRuleDescriptionContextKey() ? taintVulnerabilityFromWs.getRuleDescriptionContextKey() : null;
-    // TODO Parse actual values from stream
-    var cleanCodeAttribute = CleanCodeAttribute.TRUSTWORTHY;
-    var impacts = Map.of(
-      SoftwareQuality.RELIABILITY, ImpactSeverity.LOW,
-      SoftwareQuality.MAINTAINABILITY, ImpactSeverity.MEDIUM,
-      SoftwareQuality.SECURITY, ImpactSeverity.HIGH
-    );
+    var cleanCodeAttribute = taintVulnerabilityFromWs.hasCleanCodeAttribute() ? CleanCodeAttribute.valueOf(taintVulnerabilityFromWs.getCleanCodeAttribute().name()) : null;
```

Comment:
Not sure when the WS could return `UNKNOWN_ATTRIBUTE` (0) but I think `valueOf` would throw here. The same goes for the other 5 `valueOf` added calls below

Reply:
Following our discussion, I will make sure that we treat `UNKNOWN_XXX` enum values as `null` where appropriate.